src/app/campaign/campaign.component.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { CampaignComponent } from './campaign.component';
+import { CampaignStoreService, CampaignState, FlightState } from '../core';
+import { ReplaySubject, of } from 'rxjs';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ToastrService } from 'ngx-prx-styleguide';
+import { map } from 'rxjs/operators';
+
+describe('CampaignComponent', () => {
+  let routeId: ReplaySubject<string>;
+  let route: ActivatedRoute;
+  let router: Router;
+  let toastrService: ToastrService;
+  let campaignStoreService: CampaignStoreService;
+  let campaignState: ReplaySubject<CampaignState>;
+  let component: CampaignComponent;
+
+  function campaignFactory(attrs = {}): CampaignState {
+    return {
+      localCampaign: {
+        name: 'my campaign',
+        type: '',
+        status: '',
+        repName: '',
+        notes: '',
+        set_account_uri: '',
+        set_advertiser_uri: '',
+        ...attrs
+      },
+      changed: false,
+      valid: false,
+      flights: {}
+    };
+  }
+  function flightFactory(attrs = {}): FlightState {
+    return {
+      localFlight: { id: 123, name: 'my flight', startAt: '', endAt: '', totalGoal: 1, set_inventory_uri: '', ...attrs },
+      changed: false,
+      valid: false
+    };
+  }
+
+  beforeEach(() => {
+    routeId = new ReplaySubject(1);
+    route = { paramMap: routeId.pipe(map(id => ({ get: jest.fn(() => id) }))) } as any;
+    router = <any>{ navigate: jest.fn(), url: '/campaign/new/flight/9999' };
+    toastrService = <any>{ success: jest.fn() };
+    campaignState = new ReplaySubject(1);
+    campaignStoreService = {
```

Comment:
Right now, it looks like _most_ of the campaign store service is only used here.  So I agree, but I'm going to push that to the next (hopefully smaller) feature PR that ends up using this store.  And only partially because i'm tired of messing around with tests, and this PR needs to merge!